contracts/mainnet/connectors/basic-ERC4626/main.sol

```diff
@@ -76,51 +80,59 @@ abstract contract BasicConnector is Events, DSMath, Basic {
 	/**
 	 * @dev Mint underlying asset to ERC4626 Vault.
 	 * @notice Mints vault shares by minting exactly amount of underlying assets
-	 * @param token ERC4626 Token address.
+	 * @param vaultToken ERC4626 Token address.
 	 * @param shareAmt The amount of the share to mint. (For max: `uint256(-1)`)
-	 * @param maxTokenPerShares The max underyling token rate of mint
+	 * @param maxTokenPerShares The max underyling token rate of mint. Always in 18 decimals.
 	 * @param getId ID to retrieve amt.
 	 * @param setId ID stores the amount of tokens minted.
 	 */
-
 	function mint(
-		address token,
+		address vaultToken,
 		uint256 shareAmt,
 		uint256 maxTokenPerShares,
 		uint256 getId,
 		uint256 setId
 	) public returns (string memory _eventName, bytes memory _eventParam) {
 		uint256 _shareAmt = getUint(getId, shareAmt);
-		IERC4626 vaultTokenContract = IERC4626(token);
 
-		address _underlyingToken = vaultTokenContract.asset();
-		uint8 _vaultShareDecimal = vaultTokenContract.decimals();
+		IERC4626 vaultTokenContract = IERC4626(vaultToken);
 		TokenInterface underlyingTokenContract = TokenInterface(
-			_underlyingToken
+			vaultTokenContract.asset()
 		);
 
+		// In vault token decimals
 		_shareAmt = _shareAmt == uint256(-1)
 			? vaultTokenContract.balanceOf(address(this))
 			: _shareAmt;
 
+		// In 18 decimals
 		maxTokenPerShares = convertTo18(
-			_vaultShareDecimal,
-			wmul(maxTokenPerShares, _shareAmt)
+			vaultTokenContract.decimals(),
+			wmul(maxTokenPerShares, _shareAmt) // Returns final amount in the vault token decimals.
+		);
+
+		// In token decimals
+		maxTokenPerShares = convert18ToDec(

```

Comment:
Similar to above function we should define a new variable here as:
`_maxTokens` rather than updating `maxTokenPerShares` as it creates confusion at the end.

Reply:
Updated in https://github.com/Instadapp/dsa-connectors/pull/310/commits/2424035fa491df2194dcb5f0635bedc985d486b9